pennylane/math/quantum.py

```diff
@@ -455,6 +455,71 @@ def reduced_dm(state, indices, check_state=False, c_dtype="complex128"):
     return density_matrix
 
 
+def purity(state, indices, check_state=False, c_dtype="complex128"):
+    r"""Computes the purity from a state vector or density matrix.
+
+    .. math::
+        \gamma = Tr(\rho^2)
+
+    where :math:`\rho` is the density matrix. The purity of a normalized quantum state satisfies
+    :math:`\frac{1}{d} \leq \gamma \leq 1`, where :math:`d` is the dimension of the Hilbert space.
+    A pure state has a :math:`\gamma` of 1.
+
+    It is possible to compute the purity of a sub-system from a given state. To find the purity of
+    the overall state, include all wires in the "indices" argument.
+
+    Args:
+        state (tensor_like): ``(2**N)`` state vector or ``(2**N, 2**N)`` density matrix.
+        indices (list(int)): List of indices in the considered subsystem.
+        check_state (bool): If True, the function will check the state validity (shape and norm).
+        c_dtype (str): Complex floating point precision type.
+
+    Returns:
+        float: purity of the considered subsystem.
+
+    ** Example **
+
+    >>> x = [1, 0, 0, 1] / np.sqrt(2)
+    >>> purity(x, [0, 1])
+    1.0
+    >>> purity(x, [0])
+    0.5
+
+    >>> x = [[1 / 2, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 1 / 2]]
+    >>> purity(x, [0, 1])
+    0.5
+
+    .. seealso:: :func:`pennylane.qinfo.transforms.purity` and :func:`pennylane.purity`

```

Comment:
Does the second one refer to the measurement process? If so, we should remove it for now.